### PR TITLE
feat: use toast.promise and chirag's styling

### PR DESCRIPTION
**Pull request by harshkhandeparkar**
# Description

Please include a summary of the changes and the related issue. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

**Test Configuration**:
* Firmware version:
* Hardware:
* Toolchain:
* SDK:

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: frontend/src/utils/copyLink.ts [hunk 1]
```diff
@@ -3,12 +3,9 @@ import { toast } from 'solid-toast';
...
 
-    try {
-      await navigator.clipboard.writeText(link);
-      toast('Link copied to clipboard.');
-    } catch (e) {
-      toast('Error copying link to the clipboard.', {
-        className: "toast toast-err"
-      });
-    }
+  return toast.promise(navigator.clipboard.writeText(link), {
```
Comment (chirag-ghosh) [inline frontend/src/utils/copyLink.ts:6]: do we need to return this promise?

Comment (harshkhandeparkar) [reply frontend/src/utils/copyLink.ts:6]: Not really but doesn't hurt?

Comment (chirag-ghosh) [reply frontend/src/utils/copyLink.ts:6]: remove. give those damn registers a rest.

Comment (harshkhandeparkar) [reply frontend/src/utils/copyLink.ts:6]: yeeted.

**Review by chirag-ghosh — APPROVED** [review]

**Pull request merged**